frontend/public/app.js

```diff
@@ -56,22 +57,28 @@ class Application {
             if (event.target.status >= 200 && event.target.status < 300) {
                 this.fetchDataPeriodically();
             }
+            this.showNotificationByResponseStatus(event.target.status);
         });
 
         let data = {
             text: form.text.value,
-            stock: form.stock.value,
+            // stock: form.stock.value
             uppercase: false,
             reverse: false,
         };
 
-        let json = JSON.stringify(data);
+        if (data.text.trim().length === 0) {
```

Comment:
If we just return at the end of the `if` here we can keep this function flat. It looks a but cleaner IMO.

For example

```javascript
if (!data.text.trim().length) {
  this.setAlertMessageContent("Please enter a fruit", "fa fa-exclamation-circle", "pf-c-alert pf-m-danger pf-m-inline");
  this.showAlertForDuration(3000);
  return;
}
let json = JSON.stringify(data);
request.setRequestHeader("Content-Type", "application/json");
form.text.value = "";
```

Reply:
Yeah, sounds good to me, will refactor